File: spark-in-action/java/src/main/java/ingestion/database/MySqlToDatasetApp.java
package ingestion.database;

import java.util.Properties;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;


public class MySqlToDatasetApp {

    public static void main(String[] args) {
        MySqlToDatasetApp app = new MySqlToDatasetApp();
        app.start();
    }

    private void start() {
        SparkSession spark = SparkSession.builder()
                .appName("MySQL to Dataframe using a JDBC Connection")
                .master("local")
                .getOrCreate();

        spark.sparkContext().setLogLevel("ERROR");

        // Using properties object, which is going to be used to collect the properties
        Properties props = new Properties();
        props.put("user", "root");
        props.put("password", "mysql");
        props.put("useSLL", "false");  // Custom property (MySQL needs to be told that it won't use SSL

        Dataset<Row> df = spark.read().jdbc(
                "jdbc:mysql://localhost:3306/sakila?serverTimezone=EST",  // JDBC URL
                "actor", props);  // Table & Properties

        df = df.orderBy(df.col("last_name"));

        // Displays the dataframe and some of its metadata
        df.show(5);
        df.printSchema();;
        System.out.println("The dataframe contains " + df.count() + " records");

        spark.stop();
    }
}

File: spark-in-action/scala/src/main/scala/ingestion/database/mysql/MySqlToDatasetApp.scala
package ingestion.database.mysql

import org.apache.spark.sql.SparkSession

import java.util.Properties

object MySqlToDatasetApp {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder
      .appName("MySQL to Dataframe using a JDBC Connection")
      .master("local[*]")
      .getOrCreate

    spark.sparkContext.setLogLevel("ERROR")

    val props = new Properties
    props.put("user", "root")
    props.put("password", "mysql")
    props.put("useSSL", "false")
    props.put("allowPublicKeyRetrieval", "true")

    val mysqlUrl = "jdbc:mysql://localhost:3306/sakila"

    var df = spark.read.jdbc(mysqlUrl, "actor", props)

    df = df.orderBy(df.col("last_name"))

    df.show(5)
    df.printSchema
    println(s"The dataframe contains ${df.count} records(s)")

    spark.stop
  }
}
